pkg/traceql/engine.go

```diff
@@ -192,52 +149,52 @@ func (e *Engine) asTraceSearchMetadata(spanset *Spanset) (*tempopb.TraceSearchMe
 		}
 	}
 
-	return metadata, nil
+	return metadata
 }
 
 func unixMilliToNano(ts uint32) uint64 {
 	return uint64(ts) * 1000
 }
 
-func asAnyValue(static Static) (*common_v1.AnyValue, error) {
-	switch static.Type {
+func (s Static) asAnyValue() *common_v1.AnyValue {
+	switch s.Type {
 	case TypeInt:
 		return &common_v1.AnyValue{
 			Value: &common_v1.AnyValue_IntValue{
-				IntValue: int64(static.N),
+				IntValue: int64(s.N),
 			},
-		}, nil
+		}
 	case TypeString:
 		return &common_v1.AnyValue{
 			Value: &common_v1.AnyValue_StringValue{
-				StringValue: static.S,
+				StringValue: s.S,
 			},
-		}, nil
+		}
 	case TypeFloat:
 		return &common_v1.AnyValue{
 			Value: &common_v1.AnyValue_DoubleValue{
-				DoubleValue: static.F,
+				DoubleValue: s.F,
 			},
-		}, nil
+		}
 	case TypeBoolean:
 		return &common_v1.AnyValue{
 			Value: &common_v1.AnyValue_BoolValue{
-				BoolValue: static.B,
+				BoolValue: s.B,
 			},
-		}, nil
+		}
 	case TypeDuration:
 		return &common_v1.AnyValue{
 			Value: &common_v1.AnyValue_StringValue{
-				StringValue: static.D.String(),
+				StringValue: s.D.String(),
 			},
-		}, nil
+		}
 	case TypeStatus:
 		return &common_v1.AnyValue{
 			Value: &common_v1.AnyValue_StringValue{
-				StringValue: static.Status.String(),
+				StringValue: s.Status.String(),
 			},
-		}, nil
+		}
 	default:
-		return nil, fmt.Errorf("static has unexpected type %v", static.Type)
+		panic(fmt.Errorf("static has unexpected type %v", s.Type))
```

Comment:
Do we need to add TypeNil?

Reply:
I'm not entirely sure since I don't think we can return an attribute with value Nil. I've added it just in case, this function should never panic now.